sematic/api/endpoints/external_resources.py

```diff
@@ -0,0 +1,92 @@
+# Standard Library
+from http import HTTPStatus
+from typing import Optional
+
+# Third-party
+import flask
+
+# Sematic
+from sematic.api.app import sematic_api
+from sematic.api.endpoints.auth import authenticate
+from sematic.api.endpoints.request_parameters import jsonify_error
+from sematic.db.models.external_resource_record import ExternalResourceRecord
+from sematic.db.models.user import User
+from sematic.db.queries import (
+    get_external_resource_record,
+    get_resource_ids_by_root_id,
+    save_external_resource_record,
+    save_run_external_resource_link,
+)
+
+
+# Allow getting a list of external resource ids.
+# Q: Why not just return the actual resources?
+# A: Because we should have the endpoint for getting an external resource
+#    hooked up such that it actually interacts with the external compute
+#    to get the most up-to-date status of the records. It would be too much work
+#    for one service call to have to do that for a *list* of resources.
+# Q: Ok, then why not just return the resources without updating them for this?
+# A: Because it would be weird to have the server return stale states when queried
+#    in a list but up-to-date ones when queried one-by-one.
+@sematic_api.route("/api/v1/external_resources/ids", methods=["GET"])
```

Comment:
Why not have the root id be a path parameter? It's a simple string.
```suggestion
@sematic_api.route("/api/v1/external_resources/ids/<root_id>", methods=["GET"])
```

Reply:
it is, but that would not be a "rest"-y way to create a URL, which typically refers to resources in the path parameters rather than using path parameters as other kinds of arguments to be used. In particular, search APIs usually have the search parameters as query parameters rather than as path components. We might want `/api/v1/external_resources/ids` to be usable to search by other parameters in combination with each other and `root_id` in the future. In other words, this is the start of a search API for these resources, but with `root_id` being the only search parameter yet.

Reply:
Or in keeping with the above:
```
GET /api/v1/runs/<run_id>/external_resources
```

Reply:
K, will switch. But I will put it under the `resolutions` endpoints, since only a root run id can be used.